oauthapplication/client.go

```diff
@@ -0,0 +1,116 @@
+// Package oauthapplication provides the OAuth applications API.
+package oauthapplication
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+
+	"github.com/clerk/clerk-sdk-go/v2"
+)
+
+//go:generate go run ../cmd/gen/main.go
+
+const path = "/oauth_applications"
+
+type Client struct {
+	Backend clerk.Backend
+}
+
+func NewClient(config *clerk.ClientConfig) *Client {
+	return &Client{
+		Backend: clerk.NewBackend(&config.BackendConfig),
+	}
+}
+
+// fetches a single OAuth application by its ID.
+func (c *Client) Get(ctx context.Context, id string) (*clerk.OAuthApplication, error) {
+	path, err := clerk.JoinPath(path, id)
+	if err != nil {
+		return nil, err
+	}
+	req := clerk.NewAPIRequest(http.MethodGet, path)
+	resource := &clerk.OAuthApplication{}
+	err = c.Backend.Call(ctx, req, resource)
+	return resource, err
+}
+
+type ListParams struct {
+	clerk.APIParams
+	clerk.ListParams
+}
+
+func (params *ListParams) ToQuery() url.Values {
```

Comment:
:question: Is this method needed? Since we're not adding functionality over clerk.ListParams.ToQuery it feels redundant.

Reply:
Without I am getting the following error, 

cannot use params (variable of type *ListParams) as clerk.Params value in argument to req.SetParams: *ListParams does not implement clerk.Params (ambiguous selector *ListParams.ToQuery

Is there any other way to bypass that error?

Reply:
I see, the problem is that both ListParams and APIParams implement the ToQuery method. This reveals a bigger problem with our types.

I'd suggest leaving as is for now. :smile: